### PR TITLE
Draft for multi-party submissions related changes in the ledger API

### PR DESCRIPTION
**Pull request by rautenrieth-da**
DO NOT MERGE

This draft serves as a base for discussions around multi-party submissions.

Note: this draft only contains changes to the ledger API protobuf files. It will fail to build.

### CODE REVIEW

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto [hunk 1]
```diff
@@ -71,6 +73,23 @@ message Commands {
   // Must not be set at the same time as min_ledger_time_abs.
   // Optional
   google.protobuf.Duration min_ledger_time_rel = 11;
+
+  // Set of parties on whose behalf the command should be executed. It is up to the server to verify that the
+  // authorisation can be granted and that the connection has been authenticated for the given set of parties.
+  // This field supersedes the ``party`` field. The effective set of parties on whose behalf the command
+  // should be executed is the union of all parties listed in ``party`` and ``act_as``. This set must not be
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:80]:
```suggestion
  // should be executed is the union of all parties listed in ``party`` and ``act_as``, which must be non-empty.
```

Comment (meiersi-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:80]: and also delete the following line.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto [hunk 2]
```diff
@@ -71,6 +73,23 @@ message Commands {
...
+  // empty.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Optional
+  repeated string act_as = 12;
+
+  // Set of parties on whose behalf (in addition to all parties listed in ``act_as``) contracts can be retrieved.
+  // This affects DAML operations such as ``fetch``, ``fetchByKey``, ``lookupByKey``, ``exercise``, and ``exerciseByKey``.
+  // Note: A participant node of a DAML network can host multiple parties. Each contract present on the participant
+  // node is only visible to a subset of these parties. A command can only use contracts that are visible to at least
+  // one of the parties in ``act_as`` or ``read_as``.
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:90]: Consider adding a comment that clarifies that authorization rules for fetches are still checked.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto [hunk 3]
```diff
@@ -44,7 +44,9 @@ message Commands {
   // Party on whose behalf the command should be executed. It is up to the server to verify that the
   // authorisation can be granted and that the connection has been authenticated for that party.
   // Must be a valid PartyIdString (as described in ``value.proto``).
-  // Required
+  // Deprecated in favor of the ``act_as`` field. If both are set, then the effective list of parties on whose
+  // behalf the command should be executed is the union of all parties listed in ``party`` and ``act_as``.
+  // Optional
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:49]: TODO: adjust the text on command deduplication to explain how it works with `actAs` parties.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto [hunk 4]
```diff
@@ -112,6 +112,12 @@ message AllocatePartyRequest {
   // have to be unique.
   // Optional
   string display_name = 2;
+
+  // If set to true, then the party will be allocated in read-only mode.
+  // Read-only parties may not authorize commands. They are useful for allowing other
+  // parties to read contract data on their behalf (see ``read_as`` in ``commands.proto``).
+  // Optional
+  bool read_only = 3;
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto:120]: @rautenrieth-da : two questions
1. does this mean that read-only parties can only ever be observers?
2. for something like a `public` party, we'd need more than this wouldn't we? We'd need read-only parties to be allocated on multiple nodes?

Comment (rautenrieth-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto:120]: 1. The intention is that read-only parties can be used as `read_as` parties, but not as `act_as` parties in command submissions. As `read_as` parties are only used for visibility checks, those parties will not be used for anything in the transaction submitted to the ledger. The intention is also that any read-only party is hosted in read-write mode on another participant, which will submit transactions where the party is at least observer.
2. Yes, on a distributed ledger, a "public" party would need to be allocated on all nodes. Note that the corresponding epic (Multi-node party allocations) was deprioritized recently. On a single-node ledger, a "public" party would need to be allocated in read-write mode. 

So in short, read-only parties without multi-node party allocations don't add any value.

**Comment by rautenrieth-da** [comment]
Closing, the corresponding changes have been merged in #8125
